File: frontend/memory-game/src/types/Game.ts
export interface Card {
  id: number;
  symbol: string;
  isFlipped: boolean;
  isMatched: boolean;
  position: number;
}

export interface Game {
  id: string;
  cards: Card[];
  moves: number;
  matches: number;
  isCompleted: boolean;
  startTime: string;
  endTime?: string;
  flippedCards: number[];
  gridSize: number;
}

export interface GameMove {
  gameId: string;
  cardPosition: number;
}

export interface GameMoveResult {
  success: boolean;
  message: string;
  game?: Game;
  isMatch: boolean;
  matchedCards: number[];
}

File: api/MemoryGameApi/Models/Game.cs
namespace MemoryGameApi.Models;

public class Game
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public List<Card> Cards { get; set; } = new();
    public int Moves { get; set; } = 0;
    public int Matches { get; set; } = 0;
    public bool IsCompleted { get; set; } = false;
    public DateTime StartTime { get; set; } = DateTime.UtcNow;
    public DateTime? EndTime { get; set; }
    public List<int> FlippedCards { get; set; } = new();
    public int GridSize { get; set; } = 4; // 4x4 grid by default
}